### PR TITLE
Add shell completion support via argcomplete

### PR DESCRIPTION
**Pull request by scop**
Could replace the extras/ completions with this I think. And if argcomplete dependency isn't desired, I suppose it could be made optional.

### CODE REVIEW

File: httpie/cli/definition.py [hunk 1]
```diff
@@ -68,7 +70,8 @@
         $ http example.org hello=world   # => POST
 
     '''
-)
+).completer = ChoicesCompleter(
+    ('GET', 'HEAD', 'POST', 'PUT', 'PATCH', 'DELETE', 'OPTIONS'))
```
Comment (LuckyDenis) [inline httpie/cli/definition.py:74]: Good evening. Is there a reason why it's not in the list: CONNECT, TRACE?

Comment (scop) [reply httpie/cli/definition.py:74]: CONNECT is proxy stuff only, not sure why one would use it directly with httpie (no idea whether it'd actually work). And TRACE is such a rarity that I don't think it deserves to be there. The thinking is to offer the most commonly used choices as completions. But if there's disagreement and CONNECT and TRACE actually work, sure they could be added.

**Review by TECH90HS — APPROVED** [review]
Noce

**Comment by scop** [comment]
Rebased on current `master`.

**Pull request closed without merging**